go/vt/vtgate/evalengine/fn_numeric.go

```diff
@@ -1090,3 +1096,94 @@ func (call *builtinTruncate) typeof(env *ExpressionEnv, fields []*querypb.Field)
 		return sqltypes.Float64, f
 	}
 }
+
+type builtinCrc32 struct {
+	CallExpr
+}
+
+var _ Expr = (*builtinCrc32)(nil)
+
+func (call *builtinCrc32) eval(env *ExpressionEnv) (eval, error) {
+	arg, err := call.arg1(env)
+	if err != nil {
+		return nil, err
+	}
+	if arg == nil {
+		return nil, nil
+	}
+
+	b := evalToBinary(arg)
+	hash := crc32.ChecksumIEEE(b.bytes)
+	return newEvalUint64(uint64(hash)), nil
+}
+
+func (call *builtinCrc32) typeof(env *ExpressionEnv, fields []*querypb.Field) (sqltypes.Type, typeFlag) {
+	_, f := call.Arguments[0].typeof(env, fields)
+	return sqltypes.Uint64, f
+}
+
+type builtinConv struct {
+	CallExpr
+	collate collations.ID
+}
+
+var _ Expr = (*builtinConv)(nil)
+
+func (call *builtinConv) eval(env *ExpressionEnv) (eval, error) {
+	n, err := call.Arguments[0].eval(env)
+	if err != nil {
+		return nil, err
+	}
+	from, err := call.Arguments[1].eval(env)
+	if err != nil {
+		return nil, err
+	}
+	to, err := call.Arguments[2].eval(env)
+	if err != nil {
+		return nil, err
+	}
+
+	if n == nil || from == nil || to == nil {
+		return nil, nil
+	}
+
+	fromBase := evalToInt64(from).i
+	toBase := evalToInt64(to).i
+
+	if fromBase < -36 || (fromBase > -2 && fromBase < 2) || fromBase > 36 {
+		return nil, nil
+	}
+	if fromBase < 0 {
+		fromBase = -fromBase
+	}
+
+	if toBase < -36 || (toBase > -2 && toBase < 2) || toBase > 36 {
+		return nil, nil
+	}
+
+	var u uint64
+	if b, ok := n.(*evalBytes); ok && b.isHexOrBitLiteral() {
+		nh, _ := b.toNumericHex()
+		u = nh.u
+	} else {
+		nStr := evalToBinary(n)
+		i, err := fastparse.ParseInt64(nStr.string(), int(fromBase))
+		u = uint64(i)
+		if errors.Is(err, fastparse.ErrOverflow) {
+			u, _ = fastparse.ParseUint64(nStr.string(), int(fromBase))
+		}
+	}
+
+	var out string
+	if toBase < 0 {
+		out = strconv.FormatInt(int64(u), -int(toBase))
```

Comment:
You want `strconv.AppendInt` and `AppendUInt` here, so you don't have to copy the string into a byte slice. Then you can use `bytes.ToUpper` in-place and save yet another allocation. Same for the compiler instruction!

Reply:
@vmg `bytes.ToUpper` isn't in place, but this is a simpler conversion so added a helper that does convert in place. Also moved the rest to use byte slices as well.